woocommerce/woocommerce
# Added plugin upload and test functionality

### Changes proposed in this Pull Request:

This PR adds a new workflow to test plugins. In addition, it adds a new `upload-plugin.js` that can be used with a provided repository to upload and activate the supplied plugin. Support has also been added for both public and private repositories. In addition, this PR adds a `upload.ini` file which bumps the upload file limits in the Docker environment to support plugin uploads.

Closes https://github.com/woocommerce/woocommerce/issues/29330

### How to test the changes in this Pull Request:

1. Verify the syntax looks correct and the CI passes
2. Change directories to `plugins/woocommerce` and run `pnpm i`
3. Test the workflow using a public repository and verify the plugin is installed and activated in the test site:

```
PLUGIN_REPOSITORY="woocommerce/facebook-for-woocommerce" PLUGIN_NAME="Facebook for WooCommerce" pnpx wc-e2e test:e2e plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js
```

4. Test the workflow using a private repository and verify the plugin is installed and activated in the test site:

```
GITHUB_TOKEN="YOUR_TOKEN_HERE" PLUGIN_REPOSITORY="private/repository" PLUGIN_NAME="PLUGIN_NAME" pnpx wc-e2e test:e2e plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js
```
